keys: refactor PrettyPrint function to use safeFormatInternal

abarganier opened the pull request:
Previously the `PrettyPrint` function called the `prettyPrintInternal`
function which had similar logic to `safeFormatInternal`. This patch
modifies `safeFormatInternal` such that `PrettyPrint` can use it and
`prettyPrintInternal` has been removed. As part of this refactor,
all `ppFunc` functions have been updated to return
`redact.RedactableString`. Changes from #87952 are also included.

Resolves https://github.com/cockroachdb/cockroach/issues/88060

Release note: None

abarganier committed:
draft commit to merge pretty print and safe print of keys

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88371)
<!-- Reviewable:end -->

Santamaura committed:
keys: refactor PrettyPrint function to use safeFormatInternal

Previously the PrettyPrint function called the prettyPrintInternal
function which had similar logic to safeFormatInternal. This patch
modifies safeFormatInternal such that PrettyPrint can use it and
prettyPrintInternal has been removed. As part of this refactor,
all ppFunc functions have been updated to return
redact.RedactableString.

Resolves #88060

Release note: None
